### DISCUSSION TITLE
Could you make me see my image files on wezterm?

### DISCUSSION THREAD

**Discussion by soulinland82** [nvim-neo-tree/neo-tree.nvim#831] [Ideas]
Nvimtree allows to see image files in wezterm using this plugin ( adelarsq/image_preview.nvim ). 
But now I stop using nvimtree and now I use neotree so I couldn't use image_preview plugin and I couldn't see images.
I suggest to add these stuffs in neotree without adding preview plugin.
If it is possible, then it is very fantastic!

**Comment by pysan3** [comment]
You can do it by defining your own command.

```lua
    filesystem = {
      window = {
        mappings = {
          ["I"] = "image_wezterm", -- "<Leader>p" instead of "I" if you want the same as NvimTree
        },
      },
      commands = {
        image_wezterm = function(state)
          local node = state.tree:get_node()
          if node.type == "file" then
            require("image_preview").PreviewImage(node.path)
          end
        end,
      },
    },
```

**Reply by pysan3** [reply]
If you instead want this command for all sources (filesystem, git, buffers, ...) (tho I don't think it'll be useful in buffers), take a look at `:h neo-tree-custom-commands-global` or https://github.com/nvim-neo-tree/neo-tree.nvim/pull/846 .

**Reply by SlimShadyIAm** [reply]
This is nice! I'm wondering if there's a way to integrate it into the built in preview (`P`) feature that neo-tree has, instead of opening it as a separate buffer

**Reply by pysan3** [reply]
I requested the author and he kindly added the feature to image.nvim: https://github.com/3rd/image.nvim/issues/56.

_Theoretically_ the preview window should be showing the preview of the image now, but in real world, it seems that there are some bugs that I might have to look into (due to window focus switching happening all over the place during previewing).

At least in my environment, I got it to work with `:Neotree float`, and sometimes in `:Neotree left` but it should drastically vary by the environment.

**Reply by pysan3** [reply]
Ok, I now understand the issue and got it working locally by tweaking the code a bit.

But I'd have to look into it a bit more to implement this in a cleaner way.

Here's a POC of the working image preview :rocket:.


https://github.com/nvim-neo-tree/neo-tree.nvim/assets/41065736/b7152637-f310-43a5-b8c3-4bcba135629d

**Reply by soulinland82** [reply]
It's awesome ! Thank you so much

**Reply by pysan3** [reply]
Created an issue: https://github.com/nvim-neo-tree/neo-tree.nvim/issues/1196.
